Feature: unread status badge

franbeep opened the pull request:
Solves #2 

A few notes about this PR:
- I believe there were quite a few ways of implementing it: You could add a new attribute to every message saying if it was read or not; You could add two new variables at the conversation level, representing what how messages the user has not seen; and finally, you could use a single variable at the conversation model. Compared to the others, the last one seemed more efficiency in terms of storage, and so it was chosen. Said that, if this app needed to expand, and allow multiple users per conversation, it would require to change not only that, but remodel the entire database.
- There might be a need to solve the text overflow regarding the latest message on the conversation list box. I was not sure if that was part of the issue. If so, please do tell me.
- Since much code was changed, I expect that some of it might need improvement.
- There is a `TODO` comment in the client when you receive a message by sockets, related to an improvement that can be done about it. Since the issue #3 will be a refactoring of the sockets implementation, I decided to let it be.

Here you can see a screenshot of the app, exemplifying the unread messages features:
![Screenshot 2021-08-06 at 23-22-52 Messenger](https://user-images.githubusercontent.com/5958704/128586633-9b2e18ff-d222-466a-a77c-fe17f5fc466c.png)

Thanks!

franbeep committed:
Fixed dotenv not available at seed

franbeep committed:
Added placeholder badge to messages

franbeep committed:
Added functionality of unread messages

- New endpoint to consume when you read a message
- New modifications made mostly at the client store
- Modified conversations model to include a boolean indicating
if there is a read message or not. This will be changed to
an Integer.

franbeep committed:
Changed model unread flag to integer

- Changed the flag responsible for unread messages from
boolean to integer. With boolean you couldn't precisely
tell how many unread messages there were

franbeep committed:
Adding comments to the new additions

franbeep committed:
Changed activeConversation to a conversation

- activeConversation was the username only before. Changing it
to a conversation object helped optimize some functions

sonipranjal reviewed (commented):
Great job! I have added a few comments below; please address them and let us know on the slack when this PR is ready for a second review!

franbeep committed:
Corrected read message endpoint from POST to PATCH

franbeep committed:
Add bold to latest message at Sidebar

franbeep committed:
Added mini avatar bubble at last read message

franbeep committed:
Fixes regarding reading unread messages

FilipeGorodscy reviewed (approved):
Great job overall. I left some comments below, feel free to merge after addressing them!

franbeep committed:
Added security to /read endpoint

- Return 404 in case the message is not found
- Returns 403 if: the message is part of another conversation;
  the user is sender message sender; or the user is not part of
  the conversation
- Changed to return 200 instead of the conversation object

franbeep committed:
Improved readability of previewText and boldText

- Changed boldText to boldPreviewText, which has the same
attributes as previewText but the color and the font weight

franbeep committed:
Update client/src/store/utils/reducerFunctions.js

- Improved readability of the if condition

Co-authored-by: FilipeGorodscy <53091098+FilipeGorodscy@users.noreply.github.com>

franbeep committed:
Refactored reducer function setConvoRead

- Now it also resets the number of unread messages
- Removed reducer function to reset unread messages

franbeep committed:
Merged commits from github

- Update client/src/store/utils/reducerFunctions.js (#afaaeb9)

franbeep committed:
Merge branch 'main' into feature/read-status-message

franbeep committed:
Corrections to the merge

- Read message socket endpoint was removed during the merge because
of refactoring (and changing the filename); Now it was put back
- Changes involving activeConversation, since it changed from username
to a conversation object
- A few missing variables and imports
